### PR TITLE
[RFC] cephadm: add journald logdriver and fix compat issues

### PR DESCRIPTION
**Pull request by jschmid1**
Signed-off-by: Joshua Schmid <jschmid@suse.de>

This is a RFC pullrequest to encourage discussion around logging; especially with journald.

Feel free to add people and comments.

I'll try to keep the pr-body up to date with the latest findings.

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/cephadm/cephadm [hunk 1]
```diff
@@ -2338,7 +2339,10 @@ class CephContainer:
             priv = ['--privileged',
                     # let OSD etc read block devs that haven't been chowned
                     '--group-add=disk']
-        if self.ptrace:
+        if self.ptrace and not self.privileged:
+            # if privileged, the SYS_PTRACE cap is already added
+            # in addition, --cap-add and --privileged are mutually
+            # exclusive since podman >= 2.0
             priv.append('--cap-add=SYS_PTRACE')
```
Comment (sebastian-philipp) [inline src/cephadm/cephadm:2346]: if you make this a dedicated PR, we could merge this right away. wdyt?

Comment (jschmid1) [reply src/cephadm/cephadm:2346]: ack

Comment (jschmid1) [reply src/cephadm/cephadm]: https://github.com/ceph/ceph/pull/37133

**Review by mgfritch — COMMENTED** [review]
Adding a log driver will override what's configured by the container engine. 

If we do decide to override the log driver should we make it configurable? 

podman has a limited set of log drivers, but docker does provide for `fluentd` etc:
https://docs.docker.com/config/containers/logging/configure/#supported-logging-drivers

File: src/cephadm/cephadm [hunk 2]
```diff
@@ -2358,6 +2359,7 @@ class CephContainer:
             'run',
             '--rm',
             '--net=host',
+            '--log-driver=journald',
```
Comment (mgfritch) [inline src/cephadm/cephadm:2362]: This really only affects containers that are run in a detached `-d` state (e6792f306ab4d07251588fdca6ed3876ae3a092a)

**Comment by jschmid1** [comment]
> Adding a log driver will override what's configured by the container engine.
> If we do decide to override the log driver should we make it configurable?

Correct. The final form of this PR should probably only set the flag if specifically requested.

**Comment by mgfritch** [comment]
It's becoming rather painful to debug when the container logs are missing from journald/ It would be great to revisit this PR!
